### PR TITLE
Update R2.jl

### PR DESCRIPTION
**Pull request by farhadrclass**
Added a sigma_k tot he solver so we can access it in the callback

### CODE REVIEW

**Comment by farhadrclass** [comment]
@dpo and @tmigot would you be able to review this ?

**Review by tmigot — CHANGES_REQUESTED** [review]
Hi @farhadrclass !
This looks good to me, would you be able to add a unit test accessing this new value (and making sure it is >0) in a callback in this file https://github.com/Farhad-phd/JSOSolvers.jl/blob/R2_sigma/test/callback.jl ?

Are there other values that might be eligible?

**Comment by farhadrclass** [comment]
sure, let me add that,
for the moment we think that is the only one we need

File: src/R2.jl [hunk 1]
```diff
@@ -72,14 +72,16 @@ mutable struct R2Solver{T, V} <: AbstractOptimizationSolver
   gx::V
   cx::V
   d::V   # used for momentum term
+  σk::T
```
Comment (dpo) [inline src/R2.jl:75]:
```suggestion
  σ::T
```

**Review by dpo — CHANGES_REQUESTED** [review]

**Comment by farhadrclass** [comment]
@tmigot I added the test, and it is ready for review

File: test/callback.jl [hunk 2]
```diff
@@ -52,3 +52,22 @@ end
...
+
+@testset "Testing Solver Values" begin
+  f(x) = (x[1] - 1)^2 + 4 * (x[2] - x[1]^2)^2
+  nlp = ADNLPModel(f, [-1.2; 1.0])
+  X = [nlp.meta.x0[1]]
+  Y = [nlp.meta.x0[2]]
+  function cb(nlp, solver, stats)
+    x = solver.x
+    push!(X, x[1])
+    push!(Y, x[2])
```
Comment (tmigot) [inline test/callback.jl:64]:
```suggestion
```

File: test/callback.jl [hunk 3]
```diff
@@ -52,3 +52,22 @@ end
   end
   @test stats.iter == 8
 end
+
+@testset "Testing Solver Values" begin
+  f(x) = (x[1] - 1)^2 + 4 * (x[2] - x[1]^2)^2
+  nlp = ADNLPModel(f, [-1.2; 1.0])
+  X = [nlp.meta.x0[1]]
+  Y = [nlp.meta.x0[2]]
```
Comment (tmigot) [inline test/callback.jl:60]:
```suggestion
```

**Review by tmigot — APPROVED** [review]
I just made a couple of suggestions to simplify the tests, after that it is good for me. I let @dpo have a look if needed.

**Comment by farhadrclass** [comment]
Thank you, 
I have applied the changes, Also I noticed a small error on my end that I fixed, 

Once it is passed, what should I do to include the code in R2?

**Pull request merged**